File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      hash_map = {}

      for i,v in enumerate(nums):
        cmp = target - v 
        if cmp in hash_map:
          return [hash_map[cmp], i]
        
        hash_map[v] = i 

      return []

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i=0; i<nums.length; i++){
            int cmpl = target - nums[i];
            if (seen.containsKey(cmpl)){
                return new int[] {seen.get(cmpl),i};
            }
            seen.put(nums[i],i);
        }
        return new int[] {};
       
    }
}